code/game/machinery/harvester.dm

```diff
@@ -0,0 +1,159 @@
+
+/obj/machinery/harvester/RefreshParts()
+	interval = 0
+	var/max_time = 40
+	for(var/obj/item/stock_parts/micro_laser/L in component_parts)
+		max_time -= L.rating
+	if(max_time <= 0)
+		max_time = 1 //ungodly, but whatever
+	interval = max_time
+
+/obj/machinery/harvester/update_icon(warming_up)
+	if(warming_up)
+		icon_state = initial(icon_state)+"-charging"
+		return
+	if(state_open)
+		icon_state = initial(icon_state)+"-open"
+	else if(harvesting)
+		icon_state = initial(icon_state)+"-active"
+	else
+		icon_state = initial(icon_state)
+
+/obj/machinery/harvester/open_machine(drop = TRUE)
+	. = ..()
+	harvesting = FALSE
+
+/obj/machinery/harvester/attack_hand(mob/user)
+	if(state_open)
+		close_machine()
+	else if(!harvesting)
+		open_machine()
+
+/obj/machinery/harvester/AltClick(mob/user)
+	if(harvesting || !user || !isliving(user) || state_open)
+		return
+	if(can_harvest())
+		start_harvest()
+
+/obj/machinery/harvester/proc/can_harvest()
+	if(!powered(EQUIP) || state_open || !occupant || !iscarbon(occupant))
+		return
+	var/mob/living/carbon/C = occupant
+	if(!allow_clothing)
+		for(var/obj/item/I in C.held_items + C.get_equipped_items())
+			if(!(I.item_flags & NODROP))
+				say("Subject may not have abiotic items on.")
+				playsound(src, 'sound/machines/buzz-sigh.ogg', 30, 1)
+				return
+	if(!(MOB_ORGANIC in C.mob_biotypes))
+		say("Subject is not organic.")
+		playsound(src, 'sound/machines/buzz-sigh.ogg', 30, 1)
+		return
+	if(!allow_living && (C.stat != DEAD && !C.has_trait(TRAIT_FAKEDEATH)))     //I mean, the machines scanners arent advanced enough to tell you're alive
+		say("Subject is still alive.")
+		playsound(src, 'sound/machines/buzz-sigh.ogg', 30, 1)
+		return
+	return TRUE
+
+/obj/machinery/harvester/proc/start_harvest()
+	if(!occupant || !iscarbon(occupant))
+		return
+	var/mob/living/carbon/C = occupant
+	operation_order = list() //we're pretty much inverting C.bodyparts here, since we want chest and head last, wich are otherwise the first to go
+	for(var/i = 0 to LAZYLEN(C.bodyparts)-1) //minus one, because arrays start at 1, and otherwise we'd have the index of 6-6=0=runtime
+		operation_order += C.bodyparts[LAZYLEN(C.bodyparts)-i]
+	harvesting = TRUE
+	visible_message("<span class='notice'>The [name] begins warming up!</span>")
+	update_icon(TRUE)
+	addtimer(CALLBACK(src, .proc/harvest), interval)
+
+/obj/machinery/harvester/proc/harvest()
+	update_icon()
+	if(!harvesting || state_open || !powered(EQUIP) || !occupant || !iscarbon(occupant))
+		return
+	playsound(src, 'sound/machines/juicer.ogg', 50, 1)
+	var/mob/living/carbon/C = occupant
+	if(!LAZYLEN(operation_order)) //The list is empty, so we're done here
+		C.gib()
+		end_harvesting()
+		return
+	for(var/obj/item/bodypart/BP in operation_order) //first we do non-essential limbs
+		BP.drop_limb()
+		C.emote("scream")
+		if(BP.body_zone != "chest")
+			BP.forceMove(locate(x+1,y,z))    //Move the limbs right next to it, except chest, that's a weird one
```

Comment:
Use get_step instead

you should check the return of the locate to make sure it isn't a dense turf.

If it is you should check the other cardinal directions.

Failing all those you should use `drop_location()`